pkg/platform/model/inventory.go

```diff
@@ -364,14 +382,61 @@ func filterPlatformIDs(hostPlatform, hostArch string, platformIDs []strfmt.UUID,
 		}
 	}
 
-	if len(pids) > 0 {
-		return pids, nil
+	if len(pids) == 0 && len(fallback) == 0 {
+		return nil, &ErrNoMatchingPlatform{hostPlatform, hostArch, libcVersion}
+	} else if len(pids) == 0 {
+		pids = fallback
 	}
-	if len(fallback) > 0 {
-		return fallback, nil
+
+	if runtime.GOOS == "linux" {
+		// Sort platforms by closest matching libc version.
+		// Note: for macOS, the Platform gives a libc version based on libSystem, while sysinfo.Libc()
+		// returns the clang version, which is something different altogether. At this time, the pid
+		// list to return contains only one Platform, so sorting is not an issue and unnecessary.
+		// When it does become necessary, DX-2780 will address this.
+		// Note: the Platform does not specify libc on Windows, so this sorting is not applicable on
+		// Windows.
+		libc, err := sysinfo.Libc()
+		if err != nil {
+			return nil, errs.Wrap(err, "Unable to get system libc")
+		}
+		logging.Debug("Detected system libc: %v", libc)
+		localLibc, err := strconv.ParseFloat(libc.Version(), 32)
+		if err != nil {
+			return nil, errs.Wrap(err, "Libc version is not a number: %s", libc.Version())
+		}
+		logging.Debug("libcMap: %v", libcMap)
+		logging.Debug("Unsorted pids: %v", pids)
```

Comment:
Can we drop the debug calls? This seems useful for when you were implementing it, but shipping with this will likely just cause unnecessary noise in the log file.

If there's value in shipping it, can we combine it all into one debug call with the vital info you want?

Reply:
Okay, I've dropped the debug logging.